### PR TITLE
Afpacket softlock testing/v6

### PR DESCRIPTION
**Pull request by victorjulien**
On top of #6539.

Contains lots of diag code still, but I think the main AF_PACKET changes are close to what they should be.

### CODE REVIEW

File: src/source-af-packet.c [hunk 1]
```diff
@@ -852,65 +990,66 @@ static void AFPReleasePacket(Packet *p)
...
             break;
         }
 
-        /* Read packet from ring */
         h.raw = (((union thdr **)ptv->ring.v2)[ptv->frame_offset]);
         if (unlikely(h.raw == NULL)) {
-            /* Impossible we reach this point in normal condition, so trigger
-             * a failure in reading */
-            SCReturnInt(AFP_READ_FAILURE);
+            return AFPSuriFailure(ptv, h);
```
Comment (victorjulien) [inline src/source-af-packet.c:1006]: was `AFP_READ_FAILURE` before, should remain that way

File: src/source-af-packet.c [hunk 2]
```diff
@@ -928,55 +1067,37 @@ static int AFPReadFromRing(AFPThreadVars *ptv)
...
         if ((ptv->flags & AFP_VLAN_IN_HEADER) &&
-            (h.h2->tp_status & TP_STATUS_VLAN_VALID || h.h2->tp_vlan_tci)) {
+            (tp_status & TP_STATUS_VLAN_VALID || h.h2->tp_vlan_tci)) {
             p->vlan_id[0] = h.h2->tp_vlan_tci & 0x0fff;
             p->vlan_idx = 1;
         }
 
         if (ptv->flags & AFP_ZERO_COPY) {
             if (PacketSetData(p, (unsigned char*)h.raw + h.h2->tp_mac, h.h2->tp_snaplen) == -1) {
-                TmqhOutputPacketpool(ptv->tv, p);
```
Comment (victorjulien) [inline src/source-af-packet.c:951]: missing in new code

File: src/source-af-packet.c [hunk 3]
```diff
@@ -928,55 +1067,37 @@ static int AFPReadFromRing(AFPThreadVars *ptv)
...
         } else {
             if (PacketCopyData(p, (unsigned char*)h.raw + h.h2->tp_mac, h.h2->tp_snaplen) == -1) {
-                /* As we can possibly fail to copy the data due to invalid data, let's
-                 * skip this packet and switch to the next one.
-                 */
-                h.h2->tp_status = TP_STATUS_KERNEL;
-                if (++ptv->frame_offset >= ptv->req.v2.tp_frame_nr) {
-                    ptv->frame_offset = 0;
-                }
-                TmqhOutputPacketpool(ptv->tv, p);
```
Comment (victorjulien) [inline src/source-af-packet.c:975]: same

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk |
| .flow.wrk.flows_evicted | 5309263 | 4693221 | 113.13% |
| .flow.mgr.rows_maxlen | 115 | 539 | 21.34% | 

Pipeline 4674

**Comment by victorjulien** [comment]
Replaced by #6551

**Pull request closed without merging**